multipass-addressing/src/main/scala/com/precisely/bigdata/sample/spark/MultipassAddressing.scala

```diff
@@ -83,60 +80,83 @@ object MultipassAddressing {
+
+    var multiLineRequest: RequestAddress = input.requestAddress()
+
+    var singleLine = input.requestAddress().getStreet + " " +
+      input.requestAddress().getCity + " " +
+      input.requestAddress().getAdmin1 + " " +
+      input.requestAddress().getPostalCode
+
+    if (CollectionUtils.isNotEmpty(verifyResult)) {
+      //  create multiLine request from verify Response
+      multiLineRequest = new RequestAddress()
+      multiLineRequest.setStreet(verifyAddress.getStreet)
+      multiLineRequest.setAddressNumber(verifyAddress.getAddressNumber)
+      multiLineRequest.setAdmin1(verifyAddress.getAdmin1.getLongName)
+      multiLineRequest.setAdmin2(verifyAddress.getAdmin2.getLongName)
+      multiLineRequest.setCountry(verifyAddress.getCountry.getIsoAlpha3Code)
+      multiLineRequest.setCity(verifyAddress.getCity.getLongName)
+      multiLineRequest.setPostalCode(verifyAddress.getPostalCode)
+
+      // create singleline request from verify Response
+      singleLine = verifyAddress.getFormattedAddress + " " + verifyAddress.getCountry.getIsoAlpha3Code
+    }
+
+    val multiLineResponse = addressing.geocode(multiLineRequest, relaxedGeocoderPreferences)
     val multiLineResult = multiLineResponse.getResults
 
-    if (CollectionUtils.isNotEmpty(multiLineResult) && multiLineResult.get(0).getCustomFields
-      .get("PRECISION_CODE") != null && Seq("S5", "S6", "S7", "S8").contains(multiLineResult
-      .get(0).getCustomFields.get("PRECISION_CODE").slice(0, 2))) {
-      return multiLineResponse
+    if (CollectionUtils.isNotEmpty(multiLineResult) && multiLineResult.get(0).getLocation.getExplanation.getType != null) {
+      if ("ADDRESS_POINT".equals(multiLineResult.get(0).getLocation.getExplanation.getType.label)) {
+        return multiLineResponse
+      }
     }
 
-    //  create singleline request from multiLine Response
-    val singleLine = verifyAddress.getFormattedAddress + " " + verifyAddress.getCountry.getIsoAlpha3Code
-    val singleLineRequest: RequestAddress = new RequestAddress();
+    val singleLineRequest: RequestAddress = new RequestAddress()
     singleLineRequest.setAddressLines(Arrays.asList(singleLine))
 
-    val singleLineResult = addressing.geocode(singleLineRequest, relaxedGeocoderPreferences)
-    singleLineResult
+    val singleLineResponse = addressing.geocode(singleLineRequest, relaxedGeocoderPreferences)
+    val singleLineResult = singleLineResponse.getResults
+
+    if (CollectionUtils.isNotEmpty(singleLineResult) && singleLineResult.get(0).getLocation.getExplanation.getType != null) {
+      if ("ADDRESS_POINT".equals(singleLineResult.get(0).getLocation.getExplanation.getType.label)) {
+        return singleLineResponse
+      }
+    }
+    if (CollectionUtils.isNotEmpty(singleLineResult) && CollectionUtils.isNotEmpty(multiLineResult)) {
+      if (multiLineResult.get(0).getScore > singleLineResult.get(0).getScore) {
+        return multiLineResponse
+      }
+    }
+
+    singleLineResponse
```

Comment:
Is it possible that the multiline result got a response while the single line didn't?  If so, should we handle that here instead of just assuming we should return the single line response?

Reply:
If This can be the possible case then I will handle it and update the changes accordingly